app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt

```diff
@@ -0,0 +1,113 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AppTopBar(
+    title: String,
+    darkTheme: Boolean,
+    dynamicTheme: Boolean,
+    onSearchClicked: () -> Unit,
+    onThemeSwitchClicked: () -> Unit,
+    onDynamicSwitchClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    var expanded by remember {
+        mutableStateOf(false)
+    }
+
+    TopAppBar(
+        title = {
+            Text(
+                text = title,
+                color = MaterialTheme.colorScheme.onPrimary
+            )
+        },
+        actions = {
+            IconButton(onClick = onSearchClicked) {
+                Icon(
+                    Icons.Default.Search,
+                    contentDescription = stringResource(R.string.search_icon),
+                    tint = MaterialTheme.colorScheme.onPrimary
+                )
+            }
+            IconButton(onClick = {
+                expanded = true
+            }) {
+                Icon(
+                    imageVector = Icons.Default.MoreVert,
+                    contentDescription = stringResource(R.string.open_options),
+                    tint = MaterialTheme.colorScheme.onPrimary
+                )
+            }
+            DropdownMenu(
+                modifier = modifier.background(MaterialTheme.colorScheme.primaryContainer),
+                expanded = expanded,
+                onDismissRequest = {
+                    expanded = false
+                }
+            ) {
+                Text(
+                    text = stringResource(R.string.light_dark),
+                    modifier = Modifier.padding(start = 16.dp),
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                ThemeSwitcher(
+                    darkTheme = darkTheme,
+                    onClick = onThemeSwitchClicked
+                )
+
+                Text(
+                    text = stringResource(R.string.dynamic_theme),
+                    modifier = Modifier.padding(start = 16.dp),
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                ThemeSwitcher(
+                    darkTheme = dynamicTheme,
+                    onClick = onDynamicSwitchClicked
+                )
+            }
+        },
+        colors = TopAppBarDefaults.smallTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.primary
+        )
+    )
+}
+
+@Preview(name = "Light", showBackground = true)
+@Preview(name = "Dark", showBackground = true, uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Composable
+private fun AppTopBarPreview() {
```

Comment:
Thanks for the preview. But it is displaying the same for dark and light mode. Is this a bug?

![Screenshot_20230814_063345](https://github.com/DurtDiv3r/Dawn-of-Android/assets/11054916/c39f6735-b471-4eac-a419-538d3f1ae8e6)

Reply:
Morning, From what I've seen this is how the preview displays with Material 3 and dynamic colour enabled. I spent quite a while chasing this down to see why it does this, and when switching to Material as opposed to Material 3 dependency, and the preview displayed correctly. I did try to find an answer online as to why it does this but couldn't find anything definitive. When running the App it all displays correctly though.

Reply:
Found a fix for this after a bit of experimentation. Committed changes for both PR's.